### ISSUE TITLE
Different calendar App when clicking on date

### ISSUE THREAD

**Issue by louiskram** [MM2-0/Kvaesitso#766]
Hi, is there any possibility to choose which calendar app is openend when clicking on the date on home screen? 
I don't use the standard calendar app, so pressing on the date takes me to the wrong app.

**Comment by exstntlmsnthrp** [comment]
@louiskram It should open whatever calendar. I do not use the default calendar and don't have any problems. 

I just added a different calendar app to test. When clicking on the kvaesitso date I was given the option to choose which calendar to set as default.

**Comment by louiskram** [comment]
@exstntlmsnthrp this unfortunately doesnt work for Proton Calendar. I think it is because the app ist not registered as a calendar app, so kvaesitso doesnt let me choose if I want to use it when tapping on the date. 

It would be nice if there was the possibility to map any app to pressing on the date.

**Comment by thornySoap** [comment]
@louiskram have you already tried [app info of wrongly opening app] > [set as default] > [clear defaults]?

**Comment by louiskram** [comment]
@thornySoap the problem was that I only had the standard calendar app installed, so I guess kvaesitso assumed this was the one I wanted to open. 
After I installed another calendar app, the dialogue to choose a standard app came up. But as you can see in the screenshot, android doesn't recognize proton Calendar as a calendar app and thus doesn't let me choose it. So the problem still persists. 
![Screenshot_20240426-130450](https://github.com/MM2-0/Kvaesitso/assets/84964168/59f10711-d2bf-462e-a48b-50d556952580)

**Comment by exstntlmsnthrp** [comment]
@louiskram So I'm quite certain the problem will remain and it's not one that I believe the dev is able to fix. But @MM2-0 will have to confirm this. 

The nature of how Proton calendar works and is coded, it stays sandboxed to the Proton calendar app. It or any other app, needs to request Android calendar permissions in order to be recognized by the kvaesitso calendar widget. Some people see this as a limitation of Proton and others view it as a security feature. 

For reference, I have installed the Proton calendar app and confirmed this. It does not work with the kvaesitso widget and it does not request calendar permissions.

**Comment by louiskram** [comment]
@exstntlmsnthrp the only solution that comes to mind is to add an option to choose between all apps when tapping on the date, like it is implemented for the gestures on home screen.

**Comment by b-risk** [comment]
Yes please! I use Proton Calender and was trying to add it and almost made an issue here before I saw this.

> @exstntlmsnthrp the only solution that comes to mind is to add an option to choose between all apps when tapping on the date, like it is implemented for the gestures on home screen.

**Comment by eli-sauvage** [comment]
For anyone wondering, proton calendar now supports being set as the default calendar app in Android.

https://www.reddit.com/r/ProtonMail/s/lEZ0k8AEln

I guess this issue can be closed, I agree that it's not the launcher's if a calendar app does not advertise itself as one
